desktop/cmp/error/ErrorMessage.ts

```diff
@@ -51,29 +67,40 @@ export const [ErrorMessage, errorMessage] = hoistCmp.withFactory<ErrorMessagePro
             message,
             title,
             actionFn,
-            actionButtonProps
+            actionButtonProps,
+            detailsFn,
+            detailsButtonProps
         } = props;
 
-        if (actionFn) {
-            actionButtonProps = {...actionButtonProps, onClick: actionFn};
-        }
-
         if (isNil(error)) return null;
 
         if (!message) {
             if (isString(error)) {
-                message = error as any;
+                message = error;
             } else if (error.message) {
                 message = error.message;
             }
         }
 
+        const buttons = [];
+        if (actionFn) {
+            actionButtonProps = {...actionButtonProps, onClick: error => actionFn(error)};
+        }
+
+        if (detailsFn) {
+            detailsButtonProps = {...detailsButtonProps, onClick: error => detailsFn(error)};
+        }
+
+        if (detailsButtonProps) buttons.push(detailsButton(detailsButtonProps));
```

Comment:
Should this be in the above block? Just thinking in case a developer specifies `detailsButtonProps` but no `detailsFn`

Reply:
yea, on this and the previous item, I  guess was just trying to preserve/copy the existing semantics around action where the user actually could specify the whole thing via the actionButtonProps.  but what you are describing seems simpler to understand